app/domain/authentication/authn_jwt/id_from_decoded_token_provider.rb

```diff
@@ -0,0 +1,55 @@
+module Authentication
+  module AuthnJwt
+    # Class for providing jwt identity from the decoded token from the field specified in a secret
+    class IdentityFromDecodedTokenProvider < IdentityProviderInterface
+
+      attr_accessor :secret_fetcher
+
+      def initialize(authentication_parameters)
+        @resource_id = authentication_parameters.authenticator_resource_id
+        @decoded_token = authentication_parameters.decoded_token
+        @secret_fetcher = Conjur::FetchRequiredSecrets
+      end
+
+      def provide_jwt_identity
+        token_field_name = fetch_token_field_name
+        jwt_identity = @decoded_token[token_field_name]
+        if jwt_identity.blank?
+          raise Errors::Authentication::AuthnJwt::NoSuchFieldInToken.new(
+            token_field_name
+          )
+        end
+        jwt_identity
+      end
+
+      # Checks if variable that defined from which field in decoded token to get the id is configured
+      def identity_available?
+        resource = ::Resource[token_id_field_resource_id]
+        resource.present?
+      end
+
+      def token_id_field_resource_id
+        "#{@resource_id}/#{IDENTITY_FIELD_VARIABLE}"
+      end
+
+      private
+
+      def fetch_secret(secret_id)
+        @secret_fetcher.new.(resource_ids: [secret_id])[secret_id]
+      end
+
+      def fetch_token_field_name
+        resource_id = token_id_field_resource_id
+        begin
+          secret = fetch_secret(resource_id)
+          if secret
```

Comment:
i think it is redundant check (please validate me)

Reply:
Removed